src/components/Checkbox.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @typedef CheckboxBaseProps
+ * @prop {string} name - The `name` of the checkbox.
+ * @prop {import('preact').Ref<HTMLInputElement>} [inputRef]
+ * @prop {(checked: boolean) => void} [onToggle] - Callback when checkbox is
+ *   checked/unchecked
+ * @prop {never} [type] - Type is always 'checkbox'
+ * @prop {never} [children] - Children are not allowed
+ *
+ * @typedef {Omit<import('Preact').JSX.HTMLAttributes<HTMLInputElement>, 'onToggle'> & CheckboxBaseProps} CheckboxProps
```

Comment:
For future-dev comprehension, it might be nice to document what's going on here. I grok it, but a little bit of hand-holding would seem friendlier.